lib/matplotlib/collections.py

```diff
@@ -406,6 +410,17 @@ def draw(self, renderer):
                 gc, paths[0], combined_transform.frozen(),
                 mpath.Path(offsets), offset_trf, tuple(facecolors[0]))
         else:
+            if self._gapcolor is not None:
```

Comment:
Note that `do_single_path_optimization` is only `True` if all the lines are solid
https://github.com/matplotlib/matplotlib/blob/fc5d316a21a1d1becdd93260ebb8590c7f7ee9dc/lib/matplotlib/collections.py#L381
so `gapcolor` is only meaningful after this `else`.